### ISSUE TITLE
Cannot load model with SequentialHead 

### ISSUE THREAD

**Issue by andrewaf1** [qdrant/quaterion-models#36]
I replaced the SkipConnectionHead of my model with a SequentialHead and it trained as expected. However, when I went to load the model with `SimilarityModel.load("./encoders").to(device)`, I received the following error. 

```
model = SimilarityModel.load("./encoders").to(device)
File "/home/andrew/.cache/pypoetry/virtualenvs/imageclassifier-iYVta-lH-py3.9/lib/python3.9/site-packages/quaterion_models/model.py", line 205, in load
    head = head_class.load(head_path)
  File "/home/andrew/.cache/pypoetry/virtualenvs/imageclassifier-iYVta-lH-py3.9/lib/python3.9/site-packages/quaterion_models/heads/encoder_head.py", line 70, in load
    model = cls(**config)
TypeError: __init__() got an unexpected keyword argument 'input_embedding_size'
```

I am unsure whether this is a bug or user error. I can provide additional info if needed.

**Comment by joein** [comment]
Hi, @andrewaf1 

It is a bug, will be fixed soon, thank you!

**Comment by andrewaf1** [comment]
Thanks! I could take a look at it as well if you point me in the right direction.

**Comment by joein** [comment]
When we save a model, we store head's parameters in `config.json`.
`SequentialHead` is a bit different from the other heads and does not need parameters of its parent class, but we still put `EncoderHead` parameters into config. 

https://github.com/qdrant/quaterion-models/blob/553f517445e6ba4ca889cafc577740b4f94cfb43/quaterion_models/heads/encoder_head.py#L49-L58

When we try to load a model, we pass saved params as `**kwargs` to the model's class, but `SequentialHead` has neither `input_embedding_size`, nor `dropout` nor `**kwargs`, so we cannot pass them as key word arguments.

The fast solution would be just to add `**kwargs` to the `SequentialHead`.

You can probably expect a fix today, I will tag you.

**Comment by joein** [comment]
@monatis in `SequentialHead` we get already initialized layers and they are not compatible with our current way of saving heads

**Comment by monatis** [comment]
Will come up with a workaround shortly.

**Comment by monatis** [comment]
Hi @andrewaf1, the bug was fixed in v0.1.14, so you can upgrade your quaterion-models package:
`pip install --upgrade quaterion-models`

unfortunately there's no reliable way of restoring a sequential model from a state dict only as it simply stores tensors. So our best bet seems like saving the whole sequential model there instead of the state dict by overriding save and load methods in `SequentialHead`.
In this case you won't be able to load the currently saved model unless you reload it by defining the same layers, loading the state dict and saving it in the current format again.
 Alternatively, you can retrain it from scratch if it's not a problem. Sorry for any inconvenience.

Could you please check and verify if it solves your problem?
